### ISSUE TITLE
incorrect SSL certificate error

### ISSUE THREAD

**Issue by dennisroczek** [WikiTeam/wikiteam#265]
Might be my environment, but it is not clear to solve it at the moment:
I try to dump https://wiki.openoffice.org/wiki/Main_Page
but sadly at the end the traceroute goes to Apache with a apache.org certificate. I need either a "suppress ssl certification errors" or a solution. Thanks. (using latest Linux Mint)

**Comment by dennisroczek** [comment]
OK, I got around that problem by adding `, verify=False)` at many places where a session and a HTTP request is opened. Sadly this produces per request ~ 4 Lines but at least it is doing the dump. :-)

I will fork it und push it to my fork without the motivation of pullrequesting.

**Comment by nemobis** [comment]
dennisroczek, 11/11/2015 10:00:
> OK, I got around that problem by adding |, verify=False)| at many places
> where a session and a HTTP request is opened.

You should definitely not be forced to do this, please report to Linux 
Mint or other suitable upstream. Remember to include what version of 
python-requests and nss you have installed.

According to 
https://www.ssllabs.com/ssltest/analyze.html?d=wiki.openoffice.org their 
configuration is correct and I can connect over HTTPS both with my 
browsers and curl (fedora 21, nss-3.20.0-1.0.fc21.x86_64).

**Comment by dennisroczek** [comment]
Yes I do know. The problem is that urllib is following redirects and at the end is comparing wiki.openoffice.org with *.apache.org :-(

The traceroute shows that the server is indeed located on an apache server. For that sole wiki it doesn't interest me if the ssl check isn't performed, I need the data and I have checked the certificate.

In the end I can imagine that there might be more wikis with the same problem: provided by SSL-only having a valid certificate and sitting on a server with a different host name (+diffferent SSL certificate)

**Comment by nemobis** [comment]
@dennisroczek, do you have any other update? This issue should be tracked somewhere, but it's not quite a WikiTeam issue. We can't blindly ignore certificate warnings, so if  there are some corner cases  where a human is forced to manually add `verify=False` with judgement that's ok.

**Comment by dennisroczek** [comment]
Sorry for the late response; I'm very busy right now.

I got the backup I needed ages ago; since then I haven't done anything. I just restarted the old laptop and git pulled the new commits and still getting the "InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. For information, see https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning.


I'm now updating the system. So be patient :-D
(I guess the response will be much faster as the laptop will go on my nerves when lying on that desk ^^)
